crates/storage/store_db/in_memory.rs

```diff
@@ -89,12 +89,59 @@ impl Store {
 
 #[async_trait::async_trait]
 impl StoreEngine for Store {
-    async fn store_changes(&self, _query_plan: QueryPlan) -> Result<(), StoreError> {
-        todo!()
-    }
+    async fn store_changes_batch(&self, update_batch: UpdateBatch) -> Result<(), StoreError> {
+        let mut store = self.inner()?;
+        {
+            // store account updates
+            let mut state_trie_store = store
+                .state_trie_nodes
+                .lock()
+                .map_err(|_| StoreError::Custom("Lock Error".to_string()))?;
+            for (node_hash, node_data) in update_batch.account_updates {
+                state_trie_store.insert(node_hash, node_data);
+            }
+        }
+
+        for (hashed_address, nodes) in update_batch.storage_updates {
+            let mut addr_store = store
+                .storage_trie_nodes
+                .entry(hashed_address)
+                .or_default()
+                .lock()
+                .map_err(|_| StoreError::Custom("Lock Error".to_string()))?;
```

Comment:
Should we declare a new variant for this instead of using `StoreError::Custom`, like with `StoreError::MempoolReadLock`?